module/clock/src/mod_clock.c

```diff
@@ -30,7 +30,7 @@ static struct clock_ctx module_ctx;
 /*
  * Utility functions
  */
-
+#ifdef BUILD_HAS_NOTIFICATION
```

Comment:
Is this change required ?, I think process_response_event don't  come in notifications path

Reply:
MOD_CLOCK_API_TYPE_DRIVER_RESPONSE --> clock_driver_response_api --> clock_request_complete --> mod_clock_event_id_response --> CLOCK_EVENT_IDX_RESPONSE --> process_response_event

This is the only use that I have found and  MOD_CLOCK_API_TYPE_DRIVER_RESPONSE is under #if defined(BUILD_HAS_NOTIFICATION). 
I have compilation error for unused variable / function

Reply:
Yes, but why did you put MOD_CLOCK_API_TYPE_DRIVER_RESPONSE is under #if defined(BUILD_HAS_NOTIFICATION)?

Reply:
so I re run some test without notification and the line above and the one for MOD_CLOCK_API_TYPE_DRIVER_RESPONSE is not necessary.
Will remove it